woocommerce/woocommerce-android
# Add update reader selected option logic

Closes: #7900 


### Description


This PR adds the option to choose the frequency at which the Simulated Reader software will be updated. The user can select between Always update, meaning it will show the update dialog every time the reader is connected, never update, and update it randomly.

### Testing instructions


- Go to the Settings Screen - Developer Options
- Click on the Update Developer Options row
- Select the option from the dialog (Always, Never or Random)
- Make sure the update software dialog behaves according to the option selected: (it's never displayed every time you connect the reader if you select Never, It's always displayed every time you connect the reader if you select Always and it's randomly displayed every time you connect the reader if you select Random)

### Images/gif


Just for reference, the update dialog I'm referring to is this one:

<img width="388" alt="Screen Shot 2022-11-25 at 7 55 13 AM" src="https://user-images.githubusercontent.com/30724184/204268280-79a10273-e6ea-49bd-a42b-f9547baed186.png">

Adding a gif of the process here won't work because it's too long.





ps: adding a do not merge label because the UI to remove the update row when the simulated reader is disabled is still in progress #7964 , so I think it's better if they are merged together.
